pages/home.js

```diff
@@ -32,52 +32,98 @@ export default function Home(props) {
       })
-      .then((userid) => {
-        let roomCode = e.target.roomCode.value
-        // Update room with user added
-        //Get current usersList from room id
-        client
-          .query({ query: GET_ROOM_QUERY, variables: { roomsId: roomCode } })
-          .then((res) => {
-            let userListID = []
-            res.data.rooms[0].users.forEach((user) => {
-              userListID.push(Number(user.id))
-            })
-            userListID.push(userid)
-            updateRoom({
-              variables: {
-                updateRoomId: roomCode,
-                updateRoomUsers: userListID,
-              },
-            })
-              .then((res) =>
-                // Room created, redirecting to that room...
-                router
-                  .push({
-                    pathname: `/room/${roomCode}`,
-                  })
-                  .catch((e) => {
-                    // Room was not joined.
-                    console.log(e)
-                  })
-              )
-              .catch((e) => {
-                // User was not created.
-                console.log(e)
-              })
-          })
+
+      if (addUserRes.data.addUser.success) {
+        userid = addUserRes.data.addUser.id
+        document.cookie = `userid=${userid}`
+      } else {
+        throw 'Oops! Something went wrong'
+      }
+
+      //Get List of Users
+      const getUserListRes = await getRoomUsers({
+        variables: { roomsId: roomCode.value },
+      }).catch((e) => {
+        throw 'Oops! Something went wrong'
       })
+
+      let userListID = []
+      if (getUserListRes.data) {
+        getUserListRes.data.rooms[0].users.forEach((user) => {
+          userListID.push(Number(user.id))
+        })
+        if (!userListID.includes(userid)) {
+          userListID.push(userid)
+        }
+      } else {
+        throw 'Oops! Something went wrong'
+      }
+
+      const updateRoomRes = await updateRoom({
+        variables: {
+          updateRoomId: roomCode.value,
+          updateRoomUsers: userListID,
+          isShown: false,
+        },
+      }).catch((e) => {
+        throw 'Oops! Something went wrong'
+      })
+
+      if (!updateRoomRes.data.updateRoom.success)
+        throw 'Oops! Something went wrong'
+
+      const updateUserRes = await updatedUser({
+        variables: {
+          userInput: {
+            id: userid,
+            name: username,
+            card: undefined,
+            room: roomCode.value,
+          },
+        },
+      }).catch((e) => {
+        throw 'Oops! Something went wrong'
+      })
+
+      if (updateUserRes.data.updateUser.success) {
+        router
+          .push({
+            pathname: `/room/${roomCode.value}`,
+          })
+          .catch((e) => {
+            throw 'Oops! Something went wrong'
+          })
+      } else {
+        throw 'Oops! Something went wrong'
+      }
+    } catch (e) {
+      console.log(e)
```

Comment:
I would change the console.log to ` setJoinRoomError(e)`

This will automatically apply the ErrorLabel.